nussl/datasets/hooks.py

```diff
@@ -540,9 +548,220 @@ def __init__(self, root, mix_folder='mix_clean', mode='min', split='tr',
+            all_src.mid
+            mix.wav
+            metadata.yaml
+        Track00002:
+            ...
+
+    Items returned from this hook will be a dictionary with the keys "mix" and "sources".
+    "mix" will contain an AudioSignal which is a mix of all audio signals found in "sources".
+    "sources" will contain a dictionary, where each key is the name of a source found in `recipe`,
+    a user defined grouping of MIDI synthesized sounds in Slakh.
+    If `make_submix=False` then the value of item["sources"]["source"] is a
+    list of AudioSignals categorized "source". For example, where `dataset` is a `Slakh` Object:
+
+    >>> item = dataset[0]
+    >>> type(item["sources"]["piano"])
+    <class 'list'>
+    >>> type(item["sources"]["piano"][0])
+    <class 'nussl.core.audio_signal.AudioSignal'>
+
+    If `make_submix=True`, then each value will be a submix of stems categorized as a source.
+
+    >>> type(item["sources"]["piano"])
+    <class 'nussl.core.audio_signal.AudioSignal'>
+
+    If `midi=True`, then the returned dictionary will contain two more keys, "midi_mix" and
+    "midi_sources". The key "midi_mix" will contain a PrettyMIDI object containing the instruments
+    of all PrettyMIDI objects in "midi_sources". See http://craffel.github.io/pretty-midi/
+    for more information about PrettyMIDI objects, and how they represent MIDI files.
+    The key "midi_sources" will be a dictionary structured similarly to "sources" when `make_submix=False`,
+    where the PrettyMIDI object found at self.items[i]["midi_sources"][source][j] will be the
+    representation of the MIDI file that was synthesized to become the AudioSignal
+    object found at self.items[i]["sources"][source][j]. The MIDI files will NOT be submixed when
+    `make_submix=True`; only the audio signals are.
+
+    NOTE: If a MIDI object or audio file is specified by the `metadata.yaml` file, but is not
+    found in the midi/audio subfolders, then an empty PrettyMIDI/AudioSignal object will take its place.
+    TODO: Implement this.
+
+    It is recommended that the user uses the LibYAML bindings for better performance when using the Slakh
+    dataset. If it is installed on the user machine, it will be automatically used.
+
+    Args:
+      root (str): Path to the root of the Slakh directory
+      recipe (dict): Recipe for selecting and grouping sources in Slakh. Each key is a source type,
+        and the value is a list of midi numbers that correspond with the source. For example,
+        if you want a mix that only consists of bass and piano:
+        recipe = {
+            "bass": [32, 33, 34, 35, 36, 37, 38, 39],
+            "piano": [0, 1, 2, 3, 4, 5, 6, 7]
+        }
+        For mappings between midi numbers and instrument types, please see:
+        https://github.com/ethman/slakh-utils/blob/master/midi_inst_values/general_midi_inst_0based.txt
+      program_key (str): Key indictating midi number of an instrument. default="program_num"
```

Comment:
I'm using `program_num` because in some instrument classes, there is tons of variation. (e.g. Synth Voice in included in strings.) Maybe a better solution to this issue is if the user provides a `str` instead of list of `int`s, the source would map to the specified range of `program_num`s.

Reply:
Yeah, I think perhaps for your use case using `program_num` might be better, but I think it's more obvious what's happening if we default to using `inst_class` in the general case. 

To be clear, I'm only advocating that we change the default value for `program_num`, not the way it works.

Reply:
What do you mean by "default value for `program_num`"?  Do you mean `program_key`?

Reply:
Whoops yea, sorry. The default value for `program_key` should be `inst_class`. Got my wires crossed there! :D

Reply:
fixed